README
Managed conferencing solution using 
Asterisk, Web-MeetMe and a database-driven scheduler
by Dan Austin <dan_austin@fitawi.com>
================================================================

** Background
	It has been suggested that conferencing in general and 
MeetMe specifically represent the killer app to drive Asterisk
integrations.  While MeetMe does provide a solid foundation for
audio conferencing, there is a need for a management interface
that allows system administrators, conference administrators,
and system users to schedule, monitor and manage conferences.

** Security concerns
        It is strongly recommended that register_globals be
turned off.  The Web-MeetMe UI no longer needs this feature
turned on, and if it is turned on the system may be vulnerable
to XSS.

** Using Web-MeetMe with Asterisk 1.6 or newer
	Starting with Asterisk 1.6, the logic to enforce
conference start times, end times and resource limits have
been moved into app_meetme.  This means that this package
no longer needs or includes app_cbmysql.

** The tools

Web-MeetMe
	Web-MeetMe is a collection of PHP pages that leverage
the database for scheduling and the Asterisk Manager interface
to monitor and control active conferences.

	The scheduler currently supports these features:
		+ Conference number conflict avoidance
		+ Recurring conferences 
		  * Daily, weekly or bi-weekly 
		+ Enforces a user password if a admin password 
		  is set
		+ Seperate views for past, current and future
		  conferences
		+ Editing future conferences
		  * Includes a series of conferences
		+ Deleting past conferences
		+ CDR-like view of past conferences
		  * Includes a link to conference recordings
		+ One click features to-
		  * Extend an active conference (time)
		  * Kick out all participants
		  * Out-call to include participants 
		+ Optional early join features
		  * fuzzystart allows a caller to enter a
		    room early (in seconds, set in cbmysql.conf)
		  * earlyalert notifies a call that their 
		    conference has not yet started, if they
		    enter a valid conference number and the
		    conference is scheduled to start soon.
		    (in seconds, set in cbmysql.conf)

** Installation
* Dependencies
	-  Asterisk 1.6 or later
		!! res_odbc
		!! app_meetme configured to use RealTime
	-  Database with ODBC driver
	-  PHP 4.3 or later
		!! Reports depend on mod_gd and GD-2.0.28
		!! or later
		!! PEAR::DB (php-pear)

* Configure ODBC
	Configure /etc/odbc.ini and add a section similar to this:

[MySQL-Asterisk]
Description     = ODBC for MySQL
Driver          = MySQL
Server          = 127.0.0.1
Database        = meetme
Port            = 3306
Socket          = /tmp/mysql.sock
Option          =
Stmt            =
Trace           = yes
TraceFile       = /tmp/odbc.log


	Configure /etc/asterisk/res_odbc.conf and add a section similar
to this:

[meetme]
dsn => MySQL-Asterisk
username => <db-username>
password => <db-password>
pre-connect => yes

	Lastly add the following line to /etc/asterisk/extconfig.conf:

meetme => odbc,meetme,booking

* Setup the database
	Create a table called booking in your database.  The
table is described in ./cbmysql/db-tables-v6.txt

	You can upgrade your database from an earlier install
by using this command:

mysql < ./cbmysql/upgrade-db.txt

	There is also a use the rollback-db.txt file to
revert the database to be compatible with WMM 3.X if you
need to downgrade.

	If you will be using a SQL database to store user 
accounts instead of using an LDAP service, you can use this 
command to setup the initial administrator account to logon to 
WMM:
mysql < cbmysql/db-admin-user-create.txt


* Install Web-MeetMe
	Download the latest Web-MeetMe tgz file from
http://www.sf.net/projects/web-meetme and extract it to your webservers
document root.

	Modify the ../web-meetme/lib/defines.php to match you
database settings.

	Copy ../web-meetme/phpagi/phpagi.example.conf to 
/etc/asterisk and modify it to match the settings in your
manager.conf

********************************************************* 
You must have an account in /etc/asterisk/manager.conf
similar to this:

[MeetMe]
secret = {pick-a-password}
read = system,call,log,verbose,command,agent,user
write = system,call,log,verbose,command,agent,user

********************************************************* 


	If you intend to use the authentication functions,
it is strongly recommended that you use SSL and force all
conenctions to the Web-MeetMe pages to use HTTPS.

* LDAP or Active Directory integration
	Edit ./lib/defines and set the AUTH_TYPE to adLDAP.
Set the ADMIN_GROUP to appropriate list, the default is
"Domain Admins", but you might not want the network folks
controlling your telephony systems.  Optionally you can
chanege the browser session timeout.

	Edit ./lib/adLDAP and replace all instances of
"yourdomain" and "yourserver" with the values for your
network.

* Usage
	If all of the steps have been followed, you should be
able to open your browser, connect to the Web-MeetMe page and
schedule a conference.  A new conference start time defaults
to the time it is scheduled, so modify it if needed.  Any
required fields that are left blank will be auto-generated.
Note the conference room number and password.

	Dial the number you have assigned for conferencing, and
try the conference you have just added.  If the time and password
are correct, you will be joined to the conference.

	You can now use the Web-Meetme monitor function to manage
your conference.

** Credits
Areski for his fantastic Web-MeetMe gui which proved extremely 
easy to extend.

Mike Clark for UI enhancements and the database features to host
user accounts.


And special thanks to everyone in the Asterisk community.



Changelog
------------------------------------------------------------------------
r260 | dan_austin | 2010-02-17 11:45:15 -0800 (Wed, 17 Feb 2010) | 1 line

More unassigned variables cleaned up, fix authentication seeming to require two logons.  Make [Un]Check_All work on Firefox.  Drop unused columns from db scripts and disable verbose debugging by default
------------------------------------------------------------------------
r263 | dan_austin | 2010-02-22 10:51:33 -0800 (Mon, 22 Feb 2010) | 1 line

Really fix [Un]Check_All for Firefox
------------------------------------------------------------------------
r270 | dan_austin | 2010-07-07 13:24:32 -0700 (Wed, 07 Jul 2010) | 1 line

Russian translation
------------------------------------------------------------------------
r273 | dan_austin | 2010-10-20 15:29:22 -0700 (Wed, 20 Oct 2010) | 1 line

Add minutes used to Reports.  Alter how reports detect used conferences to reduce DB load.  Better support for 'Requests Floor' events. Alter sample extensions.conf to eliminate empty CDR records.
------------------------------------------------------------------------
r274 | dan_austin | 2010-10-20 15:35:39 -0700 (Wed, 20 Oct 2010) | 1 line

Update db scripts.
------------------------------------------------------------------------
r277 | dan_austin | 2011-03-08 10:11:31 -0800 (Tue, 08 Mar 2011) | 1 line

Update readme and remove pagination from monitor code
------------------------------------------------------------------------
r279 | dan_austin | 2011-03-16 09:48:39 -0700 (Wed, 16 Mar 2011) | 1 line

Additional logging and table configuration options in app_cbmysql
------------------------------------------------------------------------
r281 | dan_austin | 2011-10-20 09:31:55 -0700 (Thu, 20 Oct 2011) | 1 line

Merge to trunk
------------------------------------------------------------------------
r284 | dan_austin | 2011-11-28 12:16:32 -0800 (Mon, 28 Nov 2011) | 1 line

Unify conf_actions and clean-up monitor code, restore missing recording filename support in the database
------------------------------------------------------------------------

r285 | random_author | 2022-07-28 20:21:22 -0300 (Wed, 28 Jul 2022) | 99999 line

WIP to migrate to newer php version.
------------------------------------------------------------------------


daily.php
<?php
include 'lib/defines.php';
include 'lib/functions.php';

session_start (); 
if (defined ('AUTH_TYPE') && !isset ($_SESSION['auth']))
  exit;

getpost_ifset ('type');
include 'lib/header_vars.php';

if (isset ($_COOKIE['ReportNow']))
  $current = "$_COOKIE[ReportNow]";

if (!isset ($current) || strlen ($current) == 0)
  $current = date ('Y-m-d');

$currentTime = strtotime ($current);
$displayTime = date ($type == 'yearly' ? 'Y' : ($type == 'monthly'
						? 'F Y' : 'F j, Y'));
?>                  
<script language="javascript">
var curtype = '<?php echo $type ?>';
var st = 's=<?php echo $conf_sel?>&t=<?php echo $conf_past_sel?>';

$(function ()
  { setupUI ($('#reportResult'));
    changeReportType (true); });
</script>
<DIV ID="tip" CLASS="tip ui-tooltip"></DIV>
<b id="headerMsg"><?php print 'Total Conferences Scheduled and Held '. ($type == 'daily' ? 'on': 'in'); ?>&nbsp;</b>
<select NAME="year" id="year"  onChange="yearChange ()">
<?php $select_year = intval (date ('Y', $currentTime));
  $first_year = $select_year - 10;
  $last_year = min ($select_year + 10, intval (date ('Y')));
  for ($i = $last_year; $i >= $first_year; $i--)
    printf ('<OPTION %s VALUE=%d>%d', $i == $select_year ? 'SELECTED' : '',
	    $i, $i);
?>
</SELECT>
<input NAME="date" id="date" size=20 value="<?php echo $displayTime; ?>">
<input NAME="now" id="now" value="<?php echo $current;?>" style="display:none">
<br/>
<img src="report-gen.php?now=<?php echo $current ?>&type=<?php echo $type ?>" id="ReportPNG" border=1 usemap="#<?php echo $type?>Map">
<?php 
  echo '<map name="dailyMap">';
  for ($i = 0; $i <= 23; $i++)
    printf ("<area shape=\"rect\" coords=\"%d,20,%d,685\" onClick=\"downSelect ('hourly', '%02d')\" value=\"%d\">",
	    $i * 24 + 30, $i * 24 + 45, $i, $i);
  echo '</map>';
  echo '<map name="monthlyMap">';
  for ($i = 1; $i <= 31; $i++)
    printf ("<area shape=\"rect\" coords=\"%d,20,%d,685\" onClick=\"downSelect ('daily', '%02d')\" value=\"%d\">",
	    $i * 24 + 6, $i * 24 + 21, $i, $i); 
  echo '</map>';
  echo '<map name="yearlyMap">';
  for ($i = 1; $i <= 12; $i++)
    printf ("<area shape=\"rect\" coords=\"%d,20,%d,685\" onClick=\"downSelect ('monthly', '%02d')\" value=\"%d\">",
	    $i * 36 - 6, $i * 36 + 9, $i, $i); 
  echo '</map>';
?>
